server.py

```diff
@@ -0,0 +1,276 @@
+        content_type = None
+        content = None
+        status_code = 200
+        headers = {}
+        p = re.compile('/+') # Creating a regex object for the pattern of multiple contiguous forward slashes
+        requested_name = p.sub('/', requested_name) # Substituting all occurrences of the pattern with single forward slash
+        
+        if detection['type'] == 1:
+            query_start = requested_name.find('?')
+            if query_start != -1:
+                requested_name = requested_name[:query_start]
+
+            if requested_name == '/':
+                requested_name = self.run_args.index_page
+            try:
+                if requested_name[-1] == '/':
+                    requested_name = requested_name[:-1]  
+                requested_name = unquote(requested_name)
+                file_name = self.meta[requested_name]['hash']
+                content_type = self.meta[requested_name]['content_type']
+            except KeyError:
+                status_code = 404
+                
+            else:
+                path = os.path.join(self.dir, file_name)
+                if os.path.isfile(path):
+                    with open(path, 'rb') as fh:
+                        content = fh.read()
+                    if content_type:
+                        if 'text/html' in content_type:
+                            content = await self.handle_html_content(content)
+
+        elif detection['type'] == 2:
+            payload_content = detection['payload']
+            if payload_content['page']:
+                try:
+                    file_name = self.meta[payload_content['page']]['hash']
+                    content_type = self.meta[payload_content['page']]['content_type']
+                    page_path = os.path.join(self.dir, file_name)
+                    with open(page_path, encoding='utf-8') as p:
+                        content = p.read()
+                except KeyError:
+                    content = '<html><body></body></html>'
+                    content_type = r'text\html'
+
+                soup = BeautifulSoup(content, 'html.parser')
+                script_tag = soup.new_tag('div')
+                script_tag.append(BeautifulSoup(payload_content['value'], 'html.parser'))
+                soup.body.append(script_tag)
+                content = str(soup).encode()
+            else:
+                content_type = mimetypes.guess_type(payload_content['value'])[0]
+                content = payload_content['value'].encode('utf-8')
+
+            if 'headers' in payload_content:
+                headers = payload_content['headers']
+        else:
+            payload_content = detection['payload']
+            status_code = payload_content['status_code']
+
+        return (content, content_type, headers, status_code)
+
+    def start(self):
+        app = web.Application()
+        app.add_routes([web.route('*', '/{tail:.*}', self.handle_request)])
```

Comment:
could you please explain a bit this `/{tail:.*}` path?

Reply:
This is to route all http request to the handle_request handler. This is because our handler was written to handle all kind of http requests for all the paths of the specific web site (eg: '/', '/path1',  '/path/two') If we don't use this `/{tail:.*}` path, then we need to create a handler each for all the paths of all websites we are going to serve, which is practically impossible

Reply:
Check this link for more clarity :) 
https://stackoverflow.com/questions/34565705/asyncio-and-aiohttp-route-all-urls-paths-to-handler